components/common/src/templating/hooks.rs

```diff
@@ -188,25 +188,25 @@ pub trait Hook: fmt::Debug + Sized + Send {
               service_group: &str,
               pkg: &Pkg,
               svc_encrypted_password: Option<T>)
-              -> Self::ExitValue
+              -> Option<Self::ExitValue>
```

Comment:
I made this an `Option` because it lets us handle the error case for all hooks consistently and in one place. We just log out a message. Later in the code we want to know the exit value of the hook. The `None` variant becomes a type of "exit value" indicating that the hook failed to run, but we don't care why this was already logged. Maybe it would be more clear if we used a new type instead of `Option` or made a type alias? I would like to make a new type, but we lose all the ergonomics of `Option` (or would have to reimplement them as we need them).

Reply:
> so further wrapping that up in an Option (or Result, really) seems to complicate that and make it less clear.

Agreed, it does complicate it more, but it also conveys more information which is if the hook ran at all. Before, all `ExitValue`s had to implement `Default` and we would just use the default value if the hook did not run. This felt wrong as we lost useful information (if the hook even ran). To me `ExitValue` was "if the hook ran this is what its exit value was". By adding the `Option` we cover the case that it did not even run.

> That being said, I think that ExitValue itself could probably be refactored a bit to be a bit more Result-like, because I think it can currently blur the lines between a hook that ran successfully but failed,

That is what I was trying to accomplish with wrapping it in `Option`. A `None` variant Indicates it did not run. It could potnetially be expressed more clearly as a new type, but I think that type will match the semantics of `Option` very closely (if not exactly). 

> Also, ExitValue can currently be anything

Before this PR yes they could return anything, but with this PR `ExitValue` is always wrapped in an `Option`. Although, `ExitValue` could certainly be an `Option`. Leading to an `Option` wrapped in an `Option`.